lib/faraday/rack_builder.rb

```diff
@@ -20,23 +22,19 @@ class StackLocked < RuntimeError; end
     # borrowed from ActiveSupport::Dependencies::Reference &
     # ActionDispatch::MiddlewareStack::Middleware
     class Handler
-      @@constants_mutex = Mutex.new
-      @@constants = Hash.new do |h, k|
-        value = k.respond_to?(:constantize) ? k.constantize : Object.const_get(k)
-        @@constants_mutex.synchronize { h[k] = value }
-      end
+      REGISTRY = Faraday::AdapterRegistry.new
```

Comment:
I set the `AdapterRegistry` instance into a const so it is run during the initial process initialization (`require 'faraday'`), and not inside some code that could potentially be multi-threaded. 

I'm open to other approaches, as my ruby concurrency game is not strong.